lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1282,7 +1282,7 @@ protected function get_layout_styles( $block_metadata ) {
 										$spacing_rule['selector']
 									);
 								} else {
-									$format          = static::ROOT_BLOCK_SELECTOR === $selector ? '%s .%s%s' : '%s.%s%s';
+									$format          = static::ROOT_BLOCK_SELECTOR === $selector ? '%s .%s%s' : ':root %s-%s%s';
```

Comment:
Sorry if this is obvious, but I wasn't too sure: why does this need the `:root` prefix, is it to increase specificity now that it's a dash, instead of a combined classname selector? And is `:root` preferable over using `static::ROOT_BLOCK_SELECTOR`?

Reply:
I'm pretty sure I added it for specificity, but am no longer sure of the actual rule it was meant to override 😅 I can try removing it and see how we go. Good idea to use `static::ROOT_BLOCK_SELECTOR` if we do end up needing it!

Reply:
Ok, so a bit of testing shows that we actually need `:root` to override the default layout styles, because it has a slightly higher specificity than `body`:

<img width="536" alt="Screenshot 2023-02-14 at 4 55 36 pm" src="https://user-images.githubusercontent.com/8096000/218651798-0624f4fa-e592-4b94-8174-da3dc4753450.png">

Leaving it as is for now.

Reply:
Thanks for confirming!

I had another idle thought — what if we added the layout classname to the selector rather than the `:root ` prefix, so that the rule becomes `.wp-block-group-is-layout-contrained.is-layout-constrained` — I think this would then have the same level of specificity as before this PR. On the other hand, that's also an unwieldy looking selector 😅

I'm quite possibly overthinking it, but I quite like the idea of the block-level-in-global-styles rules being a kind of combination of the block's classname and the layout classname, if we can get it to work, without having to add `:root ` in order to increase the specificity. My main concern about the `:root ` prefix is that while it works, it wasn't obvious to me from looking at the rule that it's there for the extra specificity, whereas combining classnames seems a little clearer... but that could just be me (and definitely my naivety about the `:root` pseudo-class 🙂), so I don't feel too strongly about it.

Reply:
Hmm that's a good point. I _think_ `:root` has the same specificity as a regular classname, so we could do something like that, or even duplicate the composite selector like `.wp-block-group-is-layout-contrained.wp-block-group-is-layout-contrained`, so it's even more obvious that the bump is for specificity.

Reply:
I'm not sure how this is going to pan out relative to #47858 though 😅

Reply:
> I'm not sure how this is going to pan out relative to https://github.com/WordPress/gutenberg/pull/47858 though 😅

Good point! I reckon we'll likely be doing a fair bit of nudging back and forth before we land on a sweet spot 🤞 😄